lib/ace/editor.js

```diff
@@ -2830,6 +2830,31 @@ config.defineOptions(Editor.prototype, "editor", {
                 relativeNumberRenderer.detach(this);
         }
     },
+    placeholder: {
+        set: function(message) {
+            if (!this.$updatePlaceholder) {
+                this.$updatePlaceholder = function() {
+                    var value = this.renderer.$composition || this.getValue();
+                    if (value && this.renderer.placeholderNode) {
+                        this.renderer.off("afterRender", this.$updatePlaceholder);
+                        dom.removeCssClass(this.container, "ace_hasPlaceholder");
+                        this.renderer.placeholderNode.remove();
+                        this.renderer.placeholderNode = null;
+                    } else if (!value && !this.renderer.placeholderNode) {
+                        this.renderer.on("afterRender", this.$updatePlaceholder);
```

Comment:
shouldn't we call `.bind(this)` here as well?

Reply:
no, because `bind` is already called on line 2852